Sources/SwiftProtobuf/TextEncodingVisitor.swift

```diff
@@ -0,0 +1,132 @@
+// ProtobufRuntime/Sources/Protobuf/ProtobufTextEncoding.swift - Text format encoding support
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// Text format serialization engine.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+public struct TextEncodingVisitor: Visitor {
+    private var encoder: TextEncoder
+    public var result: String {return encoder.result}
+    private var nameResolver: (Int) -> String?
+
+    public init(message: Message) throws {
+        self.encoder = TextEncoder()
+        self.nameResolver =
+            ProtoNameResolvers.protoFieldNameResolver(for: message)
+        try withAbstractVisitor {(visitor: inout Visitor) in
+            try message.traverse(visitor: &visitor)
+        }
+    }
+
+    public init(message: Message, encoder: TextEncoder) throws {
+        self.encoder = encoder
+        self.nameResolver =
+            ProtoNameResolvers.protoFieldNameResolver(for: message)
+        try withAbstractVisitor {(visitor: inout Visitor) in
+            try message.traverse(visitor: &visitor)
+        }
+    }
+
+    mutating public func withAbstractVisitor(clause: (inout Visitor) throws -> ()) throws {
+        var visitor: Visitor = self
+        try clause(&visitor)
+        encoder.text = (visitor as! TextEncodingVisitor).encoder.text
+    }
+
+    mutating public func visitUnknown(bytes: Data) {
+        // TODO: Does text encoding have any provision for representing proto2 unknown field?
```

Comment:
If we do swap debugDescription to use text format (#148), we likely will want them so the info is complete.